### DISCUSSION TITLE
Allure in the json file generates the name as a list instead of a string.

### DISCUSSION THREAD

**Discussion by lambotik** [allure-framework/allure2#1925] [Bug Reports] [answered]
This is one of example: 
```"name": "test_web_table_search_added_person",
  "status": "passed",
  "attachments": [
    {
      "name": [
        "xpath",
        "//button[@id=\"addNewRecordButton\"]"
      ],
      "source": "4a06f7cb-ee61-4bca-a64e-4611e52cdaf8-attachment.png",
      "type": "image/png"
    },
    {
      "name": [
        "xpath",
        "//input[@id=\"firstName\"]"
      ],
      "source": "238a2d0a-8c89-4eaf-8003-42d4b48063da-attachment.png",
      "type": "image/png"
    },
    {
      "name": [
        "xpath",
        "//input[@id=\"lastName\"]"
      ],
      "source": "930fdeba-7266-4cab-b453-7255eb974e1d-attachment.png",
      "type": "image/png"
    },
    {
      "name": [
        "xpath",
        "//input[@id=\"userEmail\"]"
      ],
      "source": "314b1541-45af-4f5d-bd75-9ff2ff97750a-attachment.png",
      "type": "image/png"
    },
    {
      "name": [
        "xpath",
        "//input[@id=\"age\"]"
      ],
      "source": "60ac69ae-10a2-494f-8487-65deca5632cd-attachment.png",
      "type": "image/png"
    },
    {
      "name": [
        "xpath",
        "//input[@id=\"salary\"]"
      ],
      "source": "59c5fa0e-a482-49ca-9251-1415a6b53a12-attachment.png",
      "type": "image/png"
    },
    {
      "name": [
        "xpath",
        "//input[@id=\"department\"]"
      ],
      "source": "1ca0051d-b9b0-4d5d-ab86-abc83f8818a5-attachment.png",
      "type": "image/png"
    },
    {
      "name": [
        "xpath",
        "//button[@id=\"submit\"]"
      ],
      "source": "c4e4d2ed-2b16-42ca-8a0f-ddbd2cda2824-attachment.png",
      "type": "image/png"
    },
    {
      "name": [
        "xpath",
        "//input[@id=\"searchBox\"]"
      ],
      "source": "b4779d0e-b554-48fe-a3cb-049150efb56b-attachment.png",
      "type": "image/png"
    },
    {
      "name": [
        "xpath",
        "//input[@id=\"searchBox\"]"
      ],
      "source": "2f2b4037-79df-4939-a5ae-242cafb52534-attachment.png",
      "type": "image/png"
    },
    {
      "name": [
        "xpath",
        "//input[@id=\"searchBox\"]"
      ],
      "source": "14da2634-16b3-4960-8c40-360423291e35-attachment.png",
      "type": "image/png"
    },
    {
      "name": "stdout",
      "source": "80093354-703d-49dc-b806-93434f63e4bd-attachment.txt",
      "type": "text/plain"
    }
  ],
  "start": 1676999073820,
  "stop": 1676999079279,
  "uuid": "7a635f9f-67f5-4b1b-bfa8-48bfdc016196",
  "historyId": "f105571423ddc811d35c0d62129324f6",
  "testCaseId": "be8dd4c6cbcc1127535aad56972b1dbf",
  "fullName": "tests.test_elements.TestWebTable#test_web_table_search_added_person",
  "labels": [
    {
      "name": "suite",
      "value": "Elements"
    },
    {
      "name": "parentSuite",
      "value": "tests"
    },
    {
      "name": "subSuite",
      "value": "TestElements"
    },
    {
      "name": "host",
      "value": "LambotikLaptop"
    },
    {
      "name": "thread",
      "value": "4008-MainThread"
    },
    {
      "name": "framework",
      "value": "pytest"
    },
    {
      "name": "language",
      "value": "cpython3"
    },
    {
      "name": "package",
      "value": "tests.test_elements"
    }
  ]
}
```

I have described this issue in detail here: [stackoverflow](https://ru.stackoverflow.com/questions/1498727/%d0%9e%d1%88%d0%b8%d0%b1%d0%ba%d0%b0-%d0%bf%d1%80%d0%b8-%d0%b3%d0%b5%d0%bd%d0%b5%d1%80%d0%b0%d1%86%d0%b8%d0%b8-%d0%be%d1%82%d1%87%d1%91%d1%82%d0%b0-allure)

**Comment by lambotik** [comment]
Come on guys, I need help to solve this problem.

**Comment by cheshi-mantu** [comment, marked as answer]
@lambotik hi there
could you please elaborate here? How is this data being generated (code example)? You're only showing the symptoms, but the data is being generated based on your code and this is where we need to look in.

cheers

**Reply by lambotik** [reply]
https://github.com/lambotik/DemoQA

**Reply by cheshi-mantu** [reply]
could you please point to a test where you think you are facing the issue?

**Reply by cheshi-mantu** [reply]
<img width="1294" alt="image" src="https://user-images.githubusercontent.com/8607348/222761171-c9594436-0a66-4115-93be-51f845754916.png">

**Reply by cheshi-mantu** [reply]
<img width="897" alt="image" src="https://user-images.githubusercontent.com/8607348/222761496-1d57c13e-2d69-4dce-8975-e73a509ff6ba.png">

**Reply by cheshi-mantu** [reply]
<img width="945" alt="image" src="https://user-images.githubusercontent.com/8607348/222761811-9eec8e08-7dd1-4918-b40e-490fa6ffefe5.png">

**Reply by lambotik** [reply]
You can try running test_forms.py or some feature from test_interactions.py

**Reply by lambotik** [reply]
> <img alt="изображение" width="945" src="https://user-images.githubusercontent.com/8607348/222761811-9eec8e08-7dd1-4918-b40e-490fa6ffefe5.png">

this function does not play a role in the problem, even if it is removed, the problem remains, and I have an element == locator

**Reply by lambotik** [reply]
I'm sorry, but I need to run, I can answer a little later

**Reply by cheshi-mantu** [reply]
the locator itself is a tuple, not a single string

**Reply by cheshi-mantu** [reply]
you are passing the locator (tuple) to the attach screenshot
so this is not true
<img width="998" alt="image" src="https://user-images.githubusercontent.com/8607348/222766522-e5685c90-d9fc-4b19-a9c7-632af5314774.png">
so the adaptor uses the locator to create the name
<img width="501" alt="image" src="https://user-images.githubusercontent.com/8607348/222766945-166015ff-dfaa-48fc-be1d-edc1f7180040.png">

use the debugger and you'll see why the attachment's name is a list
<img width="1372" alt="image" src="https://user-images.githubusercontent.com/8607348/222767804-07aa26b4-2acb-4a37-acaf-bb60eadb072d.png">

**Reply by lambotik** [reply]
Thanks a lot. You were right. That was my fault. I apologize for the wasted time. Thank you for your help.
